app/models/company.rb

```diff
@@ -1,8 +1,18 @@
 class Company < ActiveRecord::Base
+  has_many :internships
+
   validates_presence_of :name, :contact_phone, :contact_email
   validates_uniqueness_of :name
 
-  def last_company?
-    self == Company.last
+  default_scope { order('name')}
+
+  before_save :fix_url
+
+  def fix_url
```

Comment:
i think there may be a built-in ruby method or a gem to handle this instead of writing your own logic